### ISSUE TITLE
compilation generates a "misaligned expression" typecheck error in michelson

### ISSUE THREAD

**Issue by rognierbenoit** [OCamlPro/liquidity#137]
The following (dummy) script *pb.liq*

```ocaml
[%%version 0.4]

type storage = (address, nat) map

let nat_sub (a : nat) (b : nat) (m : string) =
  match%nat a - b with
  | Plus v -> v
  | Minus _ -> Current.failwith ("nat_sub result is not a nat : " @ m)

let f (a : address) (p : address) (abcd : nat) (s : storage) =
  let v = nat_sub (0 : nat) abcd "test" in
  let r = Map.update a (Some v) s in
  Map.update p (Some (0:nat)) r

let%entry main (p : address) iStorage =
  (([] : operation list), f (Current.sender ()) p (0 : nat) iStorage)
```
generates typecheck error when compiled and type checked with commands : 

```bash
$ liquidity pb.liq
$ tezos-client typecheck script pb.liq.tz
Warning:
  
                 This is NOT the Tezos Mainnet.
  
     The node you are connecting to claims to be running on the
               Tezos Alphanet DEVELOPMENT NETWORK.
          Do NOT use your fundraiser keys on this network.
          Alphanet is a testing network, with free tokens.

From line 72 character 63 to line 74 character 66, misaligned expression
Fatal error:
  syntax error in program
```

Liquidity version is 0.401
Tezos is compiled from 63b023f25a40dbf37c7ec4f51c57ce3c3013b797 in alphanet branch

Note that changing **abcd** argument name in f function to **q**, for example, does not generate type check error.

**Comment by mebsout** [comment]
This should have been fixed in the `next` branch (by commit ff3f5f8). Let us know if that is still an issue.
(Alternatively, you can use the `liquidity-mini` of your current build which didn't have this bug).

**Comment by rognierbenoit** [comment]
Thank you, 0.402 fixes it!
